src/js/components/Grommet/Grommet.js

```diff
@@ -146,14 +145,18 @@ const Grommet = forwardRef((props, ref) => {
 
   const grommetRef = useForwardedRef(ref);
 
+  // track open FocusedContainers in a global array to manage
+  // focus event listeners for trapFocus
+  const roots = useRef([]);
   useEffect(() => {
-    if (grommetRef.current) setRoots([grommetRef.current]);
+    if (grommetRef.current) roots.current.push(grommetRef.current);
   }, [grommetRef]);
```

Comment:
I think really you'd want this to depend on `grommetRef.current`. I'm wondering if you're getting lucky because this useEffect is happening after the current gets populated, but I'm curious if there's an unmount/remount that could break it (ie. thinking ahead to why React has the StrictMode checks in new versions.)

Reply:
I get this eslint error when I try to switch it do grommetRef.current.

<img width="993" alt="Screenshot 2024-10-02 at 11 10 43 AM" src="https://github.com/user-attachments/assets/72bea3c6-766f-48cd-a8a3-0ced6764875a">